src/components/Connection/UserInfo/index.tsx

```diff
@@ -0,0 +1,58 @@
+import * as S from './style';
+import { Svg } from '@/components/Svg';
+import { UserInfoProps } from '../type';
+import { ConnectionQueries } from '@/apis/connection';
+import { toast } from 'react-toastify';
+import { PostConnectionErrorHandler } from '@/apis/connection/error';
+import { useState } from 'react';
+
+export default function UserInfo({ userInfo }: UserInfoProps) {
+  const mutationConnection = ConnectionQueries.PostConnectionQuery();
+  const [requestedUsers, setRequestedUsers] = useState<string[]>([]);
+  function connectionHandler(id: string) {
+    mutationConnection.mutate(id, {
+      onSuccess: () => {
+        setRequestedUsers([...requestedUsers, id]);
```

Comment:
해당 로직은 동기적을 동작하기 때문에 개발자의 의도와 다른 코드를 내보낼 수 있을 것 같습니다.
```suggestion
        setRequestedUsers(prev => [...prev, id]);
```
이런식으로 바꾸면 좋을 것 같습니다.